sdk/typescript/test/unit/cryptography/ed25519-keypair.test.ts

```diff
@@ -28,25 +45,26 @@ describe('ed25519-keypair', () => {
     );
   });
 
-  it('creating keypair from invalid secret key throws error', () => {
-    const secretKey = fromB64(INVALID_SECRET_KEY);
-    expect(() => {
-      Ed25519Keypair.fromSecretKey(secretKey);
-    }).toThrow('provided secretKey is invalid');
-  });
+  it('create keypair from secret key and mnemonics matches keytool', () => {
+    for (const t of TEST_CASES) {
+      // Keypair derived from mnemonic
+      const keypair = Ed25519Keypair.deriveKeypair(t[0]);
+      expect(keypair.getPublicKey().toSuiAddress()).toEqual(t[2]);
 
-  it('creating keypair from invalid secret key succeeds if validation is skipped', () => {
```

Comment:
this should not be possible anymore to have a possibility to create an invalid key AND skipping validation.